使用 algorithmicx 代替 algorithmic

at15 opened the pull request:
示例 [sjtu_thesis_algorithmicx.pdf](https://github.com/weijianwen/SJTUThesis/files/285782/sjtu_thesis_algorithmicx.pdf)

对应issue #145 
- 替换了宏包
- 默认启用算法索引, 更新了 gitignore 和 win 下的 compile.bat
- 移除了旧的example
- 移除了SWITCH的扩展和示例

at15 committed:
Fixes #145 replace algorithmic with algorithmicx

- remove old algorithmic package
- remove extension for SWITCH CASE and its example
- add algorithmicx package
- add new example
- add how to use fixed position

at15 committed:
ignore *.loa and remove them when clean up

at15 committed:
Add example for algorithm position problem

at15 commented:
switch 按理说 algorithmicx 下也可以.... 我先解决了 https://github.com/at15/SJTUThesis/issues/3 再去看下(如过二专能做完的话....

cc @gaocegege 来试下 \w/

weijianwen commented:
合并前还有几个坑要填：
- 把几个commits合并成一个；
- SWITCH的示例能想办法解决吗？
- 示例的长代码好像不会换页，页码也穿插其中了。

估计看要多少时间，如果下周五前弄完，我就正好更新下版本号。

at15 commented:
@weijianwen 
- 我是觉得commit分开可以看的清楚一些，不过可以 squash。
- switch 应该按照之前的改一下就可以。
- 长代码不换页很蛋疼，我在里面的常见问题里也说了，我试试看有没有好的解决方法。

下周二毕设答辩，假如一切顺利的话估计周三就可以都解决完了。

at15 commented:
关于换页的问题 
- http://tex.stackexchange.com/questions/70733/latex-algorithm-not-display-under-correct-section

>  Caution: algorithms cannot be cut, so if there is not enough place to put an algorithm with H option at a given spot, LATEX will place a blank and put the algorithm on the following page

单一的算法block是不能被切断的

然而 google `latex algorithmicx larger than one page` 可得
- http://tex.stackexchange.com/questions/29816/algorithm-over-2-pages 

> You should use the algorithmicx package instead and its \algstore and \algrestore commands.

at15 committed:
Use algstore algrestore to split large alg block

at15 committed:
Add doc and stackoverflow links for algo position

at15 commented:
换页的问题解决了, 不过新的问题又出现了(奥特曼的bgm响起) 算法的编号不对
![image](https://cloud.githubusercontent.com/assets/5621298/16291826/9feffd5a-393d-11e6-95af-c24681399600.png)

```
这是写在算法\ref{algo:sum_100}前面的一段话，在生成的文件里它会出现在算法\ref{algo:sum_100}前面。

\begin{algorithm}
% \begin{algorithm}[H] % 强制定位
\label{algo:sum_100}
\caption{求100以内的整数和}
\begin{algorithmic}[1] %每行显示行号
\Ensure 100以内的整数和 % 输出
\State $sum \gets 0$
\For{$i = 0 \to 100$}
    \State $sum \gets sum + i$
  \EndFor
\end{algorithmic}
\end{algorithm}
```

解决了 http://tex.stackexchange.com/questions/65993/algorithm-numbering 必须是先 `caption`再`label`

at15 committed:
Must use caption before label in algorithm

In order to generate right number
http://tex.stackexchange.com/questions/65993/algorithm-numbering
must write \caption before \label

at15 committed:
Add support for switch case statement

at15 commented:
应该是做完了，抱歉拖了这么久...... 示例 [thesis-with-updated-algo.pdf](https://github.com/weijianwen/SJTUThesis/files/329248/thesis-with-updated-algo.pdf)

另外合并PR的时候应该可以直接合并commit，https://github.com/blog/2141-squash-your-commits ，前段时间新出的功能，我还没试过.... /w\  @weijianwen

weijianwen commented:
手动合并了更改。

at15 commented:
点赞!
